lib/crypto.js

```diff
@@ -652,6 +672,14 @@ function filterDuplicates(names) {
   }).sort();
 }
 
+// timingSafeEqual reuses the timing-safe Hmac.equals function (see above) for
+// comparison of non-hash inputs, like capability URLs or authentication tokens.
+exports.timingSafeEqual = function(a, b) {
+  var key = randomBytes(32);
+  var ah = new Hmac('sha256', key).update(a);
+  return ah.validate(new Hmac('sha256', key).update(b).digest());
```

Comment:
I considered `CRYTPO_memcmp` when I started this change, but even the C implementation is not necessarily guaranteed to be constant-time in all environments. See https://noncombatant.org/2014/08/10/security-as-a-class-of-interface-guarantee/ for some extra details. This approach also avoids any non-constant-time issues in the Node->C interfaces.